sql/src/test/java/io/crate/integrationtests/SubSelectIntegrationTest.java

```diff
@@ -0,0 +1,132 @@
+        setup.setUpCharacters();
+
+        execute("select id, name " +
+                "from (select * from characters where female = true) as ch " +
+                "where name like 'Arthur'");
+        assertThat(TestingHelpers.printedTable(response.rows()),
+            is("4| Arthur\n"));
+    }
+
+    @Test
+    public void testSubSelectLimitOffset() throws Exception {
+        setup.setUpLocations();
+        execute("refresh table locations");
+
+        execute("select name " +
+                "from (select * from locations order by name limit 10 offset 5) as l " +
+                "limit 5 offset 4");
+        assertThat(TestingHelpers.printedTable(response.rows()),
+            is("End of the Galaxy\n" +
+               "Galactic Sector QQ7 Active J Gamma\n" +
+               "North West Ripple\n" +
+               "Outer Eastern Rim\n"));
+    }
+
+    @Test
+    public void testSubSelectOutputs() throws Exception {
+        execute("create table t1 (a string, i integer, x integer)");
+        ensureYellow();
+
+        execute("insert into t1 (a, i, x) values ('a', 2, 3),('b', 3, 5),('c', 5, 7),('d', 7, 11)");
+        refresh();
+
+        execute("select aa, (xxi + 1) " +
+                "from (select (xx + i) as xxi, concat(a, a) as aa " +
+                " from (select a, i, (x + x) as xx from t1) as t) as tt " +
+                "order by aa");
+
+        assertThat(TestingHelpers.printedTable(response.rows()),
+            is("aa| 9\n" +
+               "bb| 14\n" +
+               "cc| 20\n" +
+               "dd| 30\n"));
+    }
+
+    @Test
+    public void testReferenceToNestedField() throws Exception {
+        setup.groupBySetup();
+
+        execute("select gender, minAge from ( " +
+                "  select gender, min(age) as minAge from characters group by gender" +
+                ") as ch " +
+                "where gender = 'male'");
+        assertThat(TestingHelpers.printedTable(response.rows()),
+            is("male| 34\n"));
+    }
+
+    @Test
+    public void testReferenceToNestedAggregatedField() throws Exception {
+        setup.groupBySetup();
+
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("sub selects with nested aggregations are not supported");
+        execute("select gender, minAge from ( " +
+                "  select gender, min(age) as minAge from characters group by gender" +
+                ") as ch " +
+                "where (minAge * 2) < 120");
+    }
+
+    @Test
+    public void testNestedGroupByAggregation() throws Exception {
+        setup.groupBySetup();
+
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("sub selects with nested aggregations are not supported");
+        execute("select count(*) " +
+                "from (select min(age) as minAge from characters group by gender) as ch " +
+                "group by minAge");
+    }
+}
```

Comment:
pls add tests with where clauses that result in dockeys normally, in order to show that the normalization works